src/web/src/pages/Main.vue

```diff
@@ -117,18 +117,59 @@ export default {
       currentSemester: '',
       courses: [],
 
+      userID: '',
+      savedCourses: [],
+
       exportIcon: faPaperPlane,
       ICS_DAY_SHORTNAMES: ['MO', 'TU', 'WE', 'TH', 'FR', 'SA', 'SU']
     };
   },
   async created() {
     this.currentSemester = this.$route.query.semester ? this.$route.query.semester : await getDefaultSemester();
-    this.updateDataOnNewSemester();
+    
+    console.log(`Semester: ${this.currentSemester}`);
+    
+    await this.updateDataOnNewSemester();
+
+    this.userID = this.$cookies.get("userID");
```

Comment:
It would be nice if you could move the fetching of user courses into its own function and then call it in created so we don't have to reload the whole page everytime a user changes semester (which also incurs the cost of Main.vue having to be re-mounted.)

Reply:
Including it in promises.all will ensure the promises return before this is called?

Reply:
Yes, if you include it in the then() callback.

Reply:
Can't await in a non-async function, and can't declare method async

Reply:
Fixed. Please review upon push